handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -966,7 +968,12 @@ private SSLException shutdownWithError(String operation, int sslError, int error
         if (handshakeState == HandshakeState.FINISHED) {
             return new SSLException(errorString);
         }
-        return new SSLHandshakeException(errorString);
+        SSLHandshakeException exception = new SSLHandshakeException(errorString);
+        if (handshakeException != null) {
+            exception.initCause(handshakeException);
+            handshakeException = null;
+        }
+        return exception;
```

Comment:
As this change is not related to this PR I created https://github.com/netty/netty/pull/9008.